File: 148-sort-list/148-sort-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode sortList(ListNode head) {
        if(head == null) 
            return head;
        List<ListNode> arr = new ArrayList<ListNode>();
        ListNode ptr = head;
        while(ptr != null) {
            arr.add(ptr);
            ptr = ptr.next;
        }
        arr.sort((a, b) -> a.val - b.val);
        
        for(int i=0; i<arr.size()-1; i++) {
            arr.get(i).next = arr.get(i+1);
        }
        arr.get(arr.size()-1).next = null;
        return arr.get(0);
    }
}

File: 148-sort-list/148-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head == None:
            return head
        arr = []
        ptr = head
        while ptr != None:
            arr.append(ptr)
            ptr = ptr.next
        arr.sort(key = lambda x: x.val)
        
        for i in range(len(arr)-1):
            arr[i].next = arr[i+1]
        arr[len(arr)-1].next = None
        
        return arr[0]